internal/controllers/clusterextension_controller.go

```diff
@@ -251,100 +451,7 @@ func (r *ClusterExtensionReconciler) resolve(ctx context.Context, ext *ocv1alpha
 	return resultSet[0], nil
 }
 
-func (r *ClusterExtensionReconciler) installedBundle(ctx context.Context, allBundles []*catalogmetadata.Bundle, ext *ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, error) {
-	bd := &rukpakv1alpha2.BundleDeployment{}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: ext.GetName()}, bd)
-	if client.IgnoreNotFound(err) != nil {
-		return nil, err
-	}
-
-	if bd.Spec.Source.Image == nil || bd.Spec.Source.Image.Ref == "" {
-		// Bundle not yet installed
-		return nil, nil
-	}
-
-	bundleImage := bd.Spec.Source.Image.Ref
-	// find corresponding bundle for the installed content
-	resultSet := catalogfilter.Filter(allBundles, catalogfilter.And(
-		catalogfilter.WithPackageName(ext.Spec.PackageName),
-		catalogfilter.WithBundleImage(bundleImage),
-	))
-	if len(resultSet) == 0 {
-		return nil, fmt.Errorf("bundle with image %q for package %q not found in available catalogs but is currently installed via BundleDeployment %q", bundleImage, ext.Spec.PackageName, bd.Name)
-	}
-
-	sort.SliceStable(resultSet, func(i, j int) bool {
-		return catalogsort.ByVersion(resultSet[i], resultSet[j])
-	})
-
-	return resultSet[0], nil
-}
-
-func (r *ClusterExtensionReconciler) validateBundle(bundle *catalogmetadata.Bundle) error {
-	unsupportedProps := sets.New(
-		property.TypePackageRequired,
-		property.TypeGVKRequired,
-		property.TypeConstraint,
-	)
-	for i := range bundle.Properties {
-		if unsupportedProps.Has(bundle.Properties[i].Type) {
-			return fmt.Errorf(
-				"bundle %q has a dependency declared via property %q which is currently not supported",
-				bundle.Name,
-				bundle.Properties[i].Type,
-			)
-		}
-	}
-
-	return nil
-}
```

Comment:
I see it was removed but so far I do not see an equivalent of this. This is probably why you had to skip the test in `internal/controllers/clusterextension_registryv1_validation_test.go` (see my other comment there).

Reply:
This relates to the comment made here: https://github.com/operator-framework/operator-controller/pull/758/files#r1586496341. We have instead used labels on the helm release to get the installed version rather than fetching it from the bundle itself - which is what `getInstalledVersion` (https://github.com/operator-framework/operator-controller/blob/ff0c3aa6886e0af8b48dec5ae2e01e601f2f0184/internal/controllers/clusterextension_controller.go#L686) does here. There is no `catalogmetadata.Bundle` returned anymore. 

Are we missing anything?

Reply:
I think @m1kola is referring to the removal of the bundle validation logic. This should still be in place and used somewhere to validate that this is a "simple" registry+v1 bundle.

Reply:
Ah, it got missed in the rebase. A follow up commit is in progress to include it.

Reply:
Re-added https://github.com/operator-framework/operator-controller/pull/846/files#diff-a54f79b762054ccdc31e4011d28cce22af8f54d43e1fddc75745ce042f177f7eR227